src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -62,6 +95,7 @@ public MemoryPoolIterator2 ProducingStart()
 
         public void ProducingComplete(MemoryPoolIterator2 end)
         {
+            // sync writes with is immedate == true as over Stream
```

Comment:
Then it would need to return a Task which would then change the `ISocketOutput` interface; its an issue that `immedate:false` can't be passed rather than the sync write per se, as its over `Stream` rather than `ISocketOutput` which has those overloads.